catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/inputtransformer/InputTransformerProducer.java

```diff
@@ -60,66 +62,71 @@ public InputTransformerProducer(CatalogEndpoint endpoint) {
     super(endpoint);
   }
 
-  @SuppressWarnings("squid:S2095" /* The TFBOS is in a try-with-resources and will be closed */)
   protected Object transform(
       Message in, String mimeType, String transformerId, MimeTypeToTransformerMapper mapper)
       throws MimeTypeParseException, CatalogTransformerException {
-    InputStream message = in.getBody(InputStream.class);
-    if (message == null) {
-      throw new CatalogTransformerException("Message body was null; unable to generate Metacard!");
-    }
-
     MimeType derivedMimeType = null;
-    if (StringUtils.isEmpty(mimeType)) {
-      try (TemporaryFileBackedOutputStream fileBackedOutputStream =
-          new TemporaryFileBackedOutputStream()) {
-        try {
-          IOUtils.copy(message, fileBackedOutputStream);
-          derivedMimeType =
-              new MimeType(
-                  getMimeTypeFromHeader(in, fileBackedOutputStream.asByteSource().openStream()));
-        } catch (IOException e) {
-          LOGGER.debug("Failed to copy incoming inputStream message", e);
-        } finally {
-          message = fileBackedOutputStream.asByteSource().openStream();
-        }
-      } catch (IOException e) {
-        LOGGER.debug("Failed to create TemporaryFileBackedOuputStream", e);
+    try (InputStream message = in.getBody(InputStream.class);
```

Comment:
You're right that we don't care, we could continue and ingest the metacard even if it fails to close. I will add the closeQuietly().

Reply:
Going to let the exception bubble up for now so that problems can be more accurately reported and diagnosed. Eventually we'd want to fail silently and continue with the request when there is better reporting for these types of issues.